lib/Basic/Statistic.cpp

```diff
@@ -33,12 +33,16 @@ using namespace llvm::sys;
 static size_t
 getChildrenMaxResidentSetSize() {
 #if defined(HAVE_GETRUSAGE)
+#if !defined(__HAIKU__)
```

Comment:
Why is this a separate #if?

Reply:
@xwu Oops, sorry about that. Would changing it to `#if defined(HAVE_GETRUSAGE) && !defined(__HAIKU__)` and removing lines 43-45 be preferred?

Reply:
@xwu I've just refactored this change and it builds fine. Are there any more issues with this PR?

Reply:
@return I’ve only done a cursory look through the changes, but that’s all I’ve got for now :) Thanks for humoring me.